x/blob/types/payforblob.go

```diff
@@ -140,6 +172,57 @@ func CreateCommitment(namespace []byte, blobData []byte, shareVersion uint8) ([]
 	return merkle.HashFromByteSlices(subTreeRoots), nil
 }
 
+// CreateMultiShareCommitment generates a commitment over multiple blobs at
+// arbitrary points in the square. It uses the normal commitment creation
+// function per blob, and then creates a merkle root of those commitments.
+func CreateMultiShareCommitment(nsids [][]byte, blobs [][]byte, shareVersions []uint32) ([]byte, error) {
+	commitments := make([][]byte, len(nsids))
+	for i := range nsids {
+		c, err := CreateCommitment(nsids[i], blobs[i], uint8(shareVersions[i]))
+		if err != nil {
+			return nil, err
+		}
+		commitments[i] = c
+	}
+
+	return merkle.HashFromByteSlices(commitments), nil
+}
+
+// ValidatePFBComponents performs basic checks over the components of one or more PFBs.
+func ValidatePFBComponents(nsIDs [][]byte, blobs [][]byte, shareVersions []uint32) error {
+	if len(nsIDs) != len(blobs) || len(nsIDs) != len(shareVersions) {
+		return ErrMismatchedNumberOfPFBComponent.Wrapf(
+			"namespaces %d blobs %d versions %d",
+			len(nsIDs), len(blobs), len(shareVersions),
+		)
+	}
+
+	if len(blobs) == 0 {
+		return ErrNoBlobs
+	}
+
+	for _, ns := range nsIDs {
+		err := ValidateBlobNamespaceID(ns)
+		if err != nil {
+			return err
+		}
+	}
+
+	for _, blob := range blobs {
+		if len(blob) == 0 {
+			return ErrZeroBlobSize
+		}
+	}
+
+	for _, v := range shareVersions {
+		if uint8(v) != appconsts.ShareVersionZero {
```

Comment:
[optional] we may want to cast the appconst instead of the shareVersion to avoid an overflow in case a shareVersion > the size of uint8 makes it here

```suggestion
		if v != uint32(appconsts.ShareVersionZero) {
```